### DISCUSSION TITLE
printf() not printing to Interpreter for Windows WinDbg ("dbgeng locally via GADP") in Ghidra 11.0.3

### DISCUSSION THREAD

**Discussion by jsmif** [NationalSecurityAgency/ghidra#6485] [Debugger]
My expectation (and understanding from watching other videos) is that printf() should output to the Interpreter window. But that doesn't seem to be working for me in Windows w/ Ghidra 11.0.3. See below for an example of continuously hitting the same breakpoint after a printf() and not getting any output. (I also tried setting the .outmask = 0xffff to maximize output, but that didn't make any difference.)

![image](https://github.com/NationalSecurityAgency/ghidra/assets/132527732/eb1db7d7-8804-411c-8ffc-9d402666117f)

Has anyone else seen printf() not printing to the Interpreter in the debugger before?

**Comment by d-millar** [comment]
@jsmif You may want to try the new traceRMI-based debugger (vs. our original all-Java debugger).  Will require building from GitHub (or waiting for the next release).  The original debugger will, for console applications, spawn conhost and the printf() will be redirected there.  The new traceRMi versions instantiate an Interpreter which essentially is conhost (and has more natural functionality) and will intercept the printf(), as you are expecting.

**Reply by jsmif** [reply]
Thanks for your reply @d-millar. Are you saying therefore that there are no available options within the current released Ghidra to be able to see the stdout printf()s? (E.g. no optional windows or WinDbg commands that can be issued or CLI args that can be given?) 

I thought I had tested this maybe a year ago and it worked, but perhaps I am completely misremembering and I was actually using Linux not Windows?

**Reply by jsmif** [reply]
OK it looks like I was unclear about my ultimate objective. I just want to be able to see the printf() output *somewhere*; I don't *specifically* need to see it in the Interpreter window (that's just what I'm used to from Linux.)

I was launching with "dbgeng locally via GADP" because "gdb locally via GADP" is what I use on Linux. It appears that I can use "dbgeng locally via IN-VM" and then a separate foo.exe will be launched, with the printf() output visible in that window. That suffices for my purpose for now. But I will look forward to the future release where output would go to the Interpreter window just so that things behave the same across both platforms.

**Reply by d-millar** [reply]
@jsmif Generally speaking, I would say you only want to run the GADP versions if either (a) you're debugging across machines, or (b) you're worried about the debugger taking out your current VM.   Also, fingers crossed, the new versions will be better all-around, and we can deprecate both the IN-VM and GADP versions.
